Symfony/CS/DocBlock/Tag.php

```diff
@@ -26,30 +26,62 @@ class Tag
     private static $tags = array(
         'api', 'author', 'category', 'copyright', 'deprecated', 'example',
         'global', 'internal', 'license', 'link', 'method', 'package', 'param',
-        'property', 'return', 'see', 'since', 'struct', 'subpackage', 'throws',
-        'todo', 'typedef', 'uses', 'var', 'version',
+        'property', 'property-read', 'property-write', 'return', 'see',
+        'since', 'struct', 'subpackage', 'throws', 'todo', 'typedef', 'uses',
+        'var', 'version',
     );
 
     /**
-     * The tag name.
+     * All the tags with types.
      *
-     * @var string
+     * @var string[]
+     */
+    private static $tagsWithTypes = array(
+        'method', 'param', 'property', 'property-read', 'property-write',
+        'return', 'throws', 'type', 'var',
+    );
+
+    /**
+     * The line containing the tag.
+     *
+     * @var Line
+     */
+    private $line;
+
+    /**
+     * The cached tag name.
+     *
+     * @var string|null
      */
     private $name;
 
     /**
+     * The cached types content.
+     *
+     * @var string|null
+     */
+    private $typesContent;
+
+    /**
      * Create a new tag instance.
      *
-     * @param string $content
+     * @param Line $line
      */
-    public function __construct($content)
+    public function __construct(Line $line)
```

Comment:
Ahh, wait, in reflection, I've put these methods on the wrong class. They should have gone on the annotation class, lol. I'll need to revert some of this. ;)